src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerConfigOptionsProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Provide options from an analyzer config file keyed on a source file.
+    /// </summary>
+    public abstract class AnalyzerConfigOptionsProvider
+    {
+        /// <summary>
+        /// Get options for a given <paramref name="tree"/>.
+        /// </summary>
+        public abstract AnalyzerConfigPropertyMap GetOptions(SyntaxTree tree);
```

Comment:
1. the naming here is strange.  The outer calss is called AnalyzerConfigOptionsProvider, but it doesn't provide AnalyzerConfigOptions.  Instead, it provides AnalyzerConfigPropertyMaps.  THis is doubly strange given that the methods are called GetOptions.

2. WHy is this an abstract class.  Is there something about the impl that all implementations would need to shared?  It feels like an interface would be better.  Especially if this is intended for IDE types that may want to implement this functionality on a type that already has its own inheritance chain.

Reply:
We may want to add helpers to this class later, especially helpers for parsing simple value strings like "true" and "false" and getting a boolean back.